src/UnleashBuilder.php

```diff
@@ -525,14 +534,28 @@ public function build(): Unleash
         $this->initializeServices($metricsHandler, $dependencyContainer);
         $this->initializeServices($variantHandler, $dependencyContainer);
 
-        return new DefaultUnleash(
-            $this->strategies,
-            $repository,
-            $registrationService,
-            $configuration,
-            $metricsHandler,
-            $variantHandler,
-        );
+        if ($this->proxyKey !== null) {
+            $configuration = $configuration->setProxyKey($this->proxyKey);
```

Comment:
Configuration is a stateful object, no need to reassign:

```suggestion
            $configuration->setProxyKey($this->proxyKey);
```